packages/aws-cdk/test/integ/cli/aws-helpers.ts

```diff
@@ -96,29 +97,44 @@ export async function deleteStacks(...stackNames: string[]) {
     });
   }
 
-  await retry(`Deleting ${stackNames}`, afterSeconds(300), async () => {
+  await retry(`Deleting ${stackNames}`, retry.forSeconds(600), async () => {
     for (const stackName of stackNames) {
-      if (await stackExists(stackName)) {
+      const status = await stackStatus(stackName);
+      if (status !== undefined && status.endsWith('_FAILED')) {
+        throw retry.abort(new Error(`'${stackName}' is in state '${status}'`));
+      }
+      if (status !== undefined) {
```

Comment:
Do we have to repeat `status !=== undefined` twice? Something like this instead?

```ts
      if (status !== undefined) {
        if (status.endsWith('_FAILED')) {
          throw retry.abort(new Error(`'${stackName}' is in state '${status}'`));
        }
        throw new Error(`Delete of '${stackName}' not complete yet`);
      }
```

I would probably even throw an `else` there before the second `throw`.